src/TLuaInterpreter.cpp

```diff
@@ -4842,6 +4842,9 @@ int TLuaInterpreter::searchRoom(lua_State* L)
         lua_newtable(L);
         QList<int> roomIdsFound;
         for (auto pR : roomList) {
+            if (!pR) {
```

Comment:
This validates that the given room pointer is valid - so its Id is added to the list `roomIdsFound` ...